test/unit/association_preloader_test.rb

```diff
@@ -0,0 +1,130 @@
+require 'test_helper'
+
+class AssociationPreloaderTest < MiniTest::Test
+  
+  def setup
+    @users =  [ Account::V1::User.new( id: 1, permission_ids: [1, 2] ), 
+                Account::V1::User.new( id: 2, permission_ids: [3, 4] ),
+              ]
+
+    @attrs =  [ Account::Client::Attribute.new( id: 1, user_id: 1 ),
+                Account::Client::Attribute.new( id: 2, user_id: 1 ),
+                Account::Client::Attribute.new( id: 3, user_id: 2 ),
+                Account::Client::Attribute.new( id: 4, user_id: 1 ),
+              ]
+  end
+
+  def test_belongs_to_bulk_query
+    users = (1..10).map do |id|
+      Account::V1::User.new({id: id, associated_website_user_id: id})
+    end
+
+    bt = Account::V1::User._associations[:associated_website_user]
+
+    bt = JsonApiResource::Associations::Preloaders::BelongsToPreloader.new(bt)
+
+    assert bt.bulk_query(users).keys.include? :id
+    assert_equal [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], bt.bulk_query(users)[:id]
+  end
+
+  def test_has_one_bulk_query
+    users = (1..10).map do |id|
+      Account::V1::User.new({id: id})
+    end
+
+    ho = Account::V1::User._associations[:address]
+    ho = JsonApiResource::Associations::Preloaders::HasOnePreloader.new(ho)
+
+    assert ho.bulk_query(users).keys.include? :user_id
+    assert_equal [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], ho.bulk_query(users)[:user_id]
```

Comment:
Bulk query returns the generated association query. So like for `Preloader.preload lawyers, :reviews` it will generate `{ lawyer_id: lawyer.map &:id }`